src/query/dblink_scan.c

```diff
@@ -543,8 +547,87 @@ dblink_bind_param (int stmt_handle, VAL_DESCR * vd, DBLINK_HOST_VARS * host_vars
 }
 
 int
-dblink_execute_query (struct access_spec_node *spec, VAL_DESCR * vd, DBLINK_HOST_VARS * host_vars)
+dblink_end_tran (THREAD_ENTRY * thread_p, bool is_abort)
+{
+  int rc = NO_ERROR;
+  T_CCI_ERROR err_buf;
+  DBLINK_CONN_ENTRY *prev, *dblink = thread_p->dblink_entry;
+
+  while (dblink)
+    {
+      rc = cci_end_tran (dblink->conn_handle, is_abort ? CCI_TRAN_ROLLBACK : CCI_TRAN_COMMIT, &err_buf);
+      if (rc < 0)
+	{
+	  is_abort = true;
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_DBLINK_TRAN, 1, err_buf.err_msg);
+	}
+
+      rc = cci_disconnect (dblink->conn_handle, &err_buf);
+      if (rc < 0)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_DBLINK_TRAN, 1, err_buf.err_msg);
+	}
+
+      /* delete dblink_entry */
+      prev = dblink;
+      dblink = dblink->next;
+
+      free (prev);
+    }
+
+  thread_p->dblink_entry = NULL;
+
+  return rc;
+}
+
+static int
+dblink_find_conn_handle (THREAD_ENTRY * thread_p, char *conn_url, char *user_name, char *password)
+{
+  DBLINK_CONN_ENTRY *dblink = thread_p->dblink_entry;
+
+  while (dblink)
+    {
+      if (!strcmp ((char *) dblink->conn_url, conn_url) && !strcmp ((char *) dblink->user_name, user_name)
+	  && !strcmp ((char *) dblink->password, password))
```

Comment:
```suggestion
      if (!strcmp (dblink->conn_url, conn_url) && !strcmp (dblink->user_name, user_name)
	  && !strcmp (dblink->password, password))
```

혹시 char*로 casting 하시는 이유가 따로 있으실까요?

Reply:
strcmp는 char *를 argument로 받는데, dblink->conn_url이나 password 등은 const char *입니다. Warning이 발생할 수 있어서 char *로 캐스팅합니다.

Reply:
```
typedef struct dblink_conn_entry DBLINK_CONN_ENTRY;
struct dblink_conn_entry
{
  int conn_handle;
  char conn_url[MAX_LEN_CONNECTION_URL + 1];
  char user_name[DB_MAX_USER_LENGTH + 1];
  char *password[DB_MAX_PASSWORD_LENGTH + 1];

  DBLINK_CONN_ENTRY *next;
};
```
구조체는 char *로 선언되어 있는데, 혹시 어디 부분에서 const char*로 변경되는지 알 수 있을까요?

Reply:
char array는 const char *와 동일한 것으로 간주합니다.